src/main.rs

```diff
@@ -63,6 +63,8 @@ enum TransportProtocol {
 #[derive(Deserialize)]
 #[serde(deny_unknown_fields)]
 struct Config {
+    #[serde(default)]
+    brightness: Option<Brightness>,
```

Comment:
Since the `Brightness` enum implements `Default` there is no need to wrap the brightness in an `Option`.

And if you are already at this location: Super nit-picky, but I prefer a blank line between struct fields if they have attributes.